src/DecayTimeFinderAction.cpp

```diff
@@ -26,11 +28,31 @@ G4ClassificationOfNewTrack DecayTimeFinderAction::ClassifyNewTrack( const G4Trac
   if ( m_firstDecay == 0.0 && track->GetGlobalTime() != 0.0 )
   {
     m_firstDecay = track->GetGlobalTime();
+    
     //std::cout << "SET DECAY TIME" << std::endl;
+
+    //Get the position of the radioactive decay
+    m_radDecayX = track->GetPosition().x();
+    m_radDecayY = track->GetPosition().y();
+    m_radDecayZ = track->GetPosition().z();
   }
 
+  // Get the position of the electron-positron annihilation
+  if (track->GetParticleDefinition()->GetParticleName() == "gamma" && track->GetParentID() == 4){
```

Comment:
My idea was that this way it makes sure that it is the positron coming from the beta+ decay and not any other although now looking at the interactions it seems that there is only one positron and only electrons appear more than once for all tracers. When writing that part I also verified that for all tested tracers the ID for the positron from the beta decay is for sure 4. Do you think it is still better to change this condition?

Reply:
Well, I think that testing it and seeing that it is always 4 is good but still no guarantee that it will work in future. What we are definitely looking for is gammas that come from positron annihilation. We can explicitly specify this. The fact that the particle ID always happens to be 4 is nice, but it's indirect and potentially fragile.

Reply:
As discussed elsewhere the only information about the parent particle was its ID so I could not ask for the parent particle name in the condition. It is now modified to look for the specific process name (annihil) to ensure the correct gammas are selected.